application/src/main/java/org/togetherjava/tjbot/commands/tags/TagCommand.java

```diff
@@ -46,20 +50,35 @@ public void onSlashCommand(SlashCommandInteractionEvent event) {
         String id = Objects.requireNonNull(event.getOption(ID_OPTION)).getAsString();
         OptionMapping replyToUserOption = event.getOption(REPLY_TO_USER_OPTION);
 
-        if (tagSystem.handleIsUnknownTag(id, event)) {
+        if (tagSystem.handleIsUnknownTag(id, event, super.getComponentIdGenerator(),
+                replyToUserOption)) {
             return;
         }
 
-        ReplyCallbackAction message = event
-            .replyEmbeds(new EmbedBuilder().setDescription(tagSystem.getTag(id).orElseThrow())
-                .setFooter(event.getUser().getName() + " • used " + event.getCommandString())
+        sendTagReply(event, event.getUser().getName(), id, event.getCommandString(),
+                replyToUserOption == null ? null : replyToUserOption.getAsUser().getAsMention());
+    }
+
+    @Override
+    public void onButtonClick(ButtonInteractionEvent event, List<String> args) {
+        sendTagReply(event, event.getUser().getName(), args.get(0), null, args.get(1));
+    }
+
+    /**
+     * Sends the reply for a successfull /tag use (i.e. the given tag exists)
+     */
+    private void sendTagReply(IReplyCallback callback, String userName, String tag,
+            @Nullable String commandString, @Nullable String replyToUser) {
```

Comment:
I have some trouble with some variable names. Some of them dont really say what they are about.

* `callback` - callback of what? I think it would be easier and more idiomatic if you just call it `event` (even if the type is a level higher).
* its easy to confuse the two users that play a role here, since both are called "user". maybe keep `userName`, but rename `replyToUser` to `replyTargetUser`. if u want, rename `userName` to `invokerUserName` maybe, then its super clear.